src/main/java/org/springframework/integration/kafka/util/TopicUtils.java

```diff
@@ -0,0 +1,157 @@
+	 * @param topicName the name of the topic
+	 * @param numPartitions the number of partitions for the topic
+	 * @param topicConfig additional topic configuration properties
+	 * @param replicaAssignment the mapping of partitions to broker
+	 *
+	 * @return {@link TopicMetadata} information for the topic
+	 */
+	public static TopicMetadata ensureTopicCreated(final ZkClient zkClient, final String topicName, final int numPartitions,
+			Properties topicConfig, Map<Object, Seq<Object>> replicaAssignment) {
+		AdminUtils.createOrUpdateTopicPartitionAssignmentPathInZK(zkClient, topicName, replicaAssignment, topicConfig,
+				true);
+
+		RetryTemplate retryTemplate = new RetryTemplate();
+
+		CompositeRetryPolicy policy = new CompositeRetryPolicy();
+		TimeoutRetryPolicy timeoutRetryPolicy = new TimeoutRetryPolicy();
+		timeoutRetryPolicy.setTimeout(METADATA_VERIFICATION_TIMEOUT);
+		SimpleRetryPolicy simpleRetryPolicy = new SimpleRetryPolicy();
+		simpleRetryPolicy.setMaxAttempts(METADATA_VERIFICATION_RETRY_ATTEMPTS);
+		policy.setPolicies(new RetryPolicy[] {timeoutRetryPolicy, simpleRetryPolicy});
+		retryTemplate.setRetryPolicy(policy);
+
+		ExponentialBackOffPolicy backOffPolicy = new ExponentialBackOffPolicy();
+		backOffPolicy.setInitialInterval(METADATA_VERIFICATION_RETRY_INITIAL_INTERVAL);
+		backOffPolicy.setMultiplier(METADATA_VERIFICATION_RETRY_BACKOFF_MULTIPLIER);
+		backOffPolicy.setMaxInterval(METADATA_VERIFICATION_MAX_INTERVAL);
+		retryTemplate.setBackOffPolicy(backOffPolicy);
+
+		try {
+			TopicMetadata topicMetadata = retryTemplate.execute(new RetryCallback<TopicMetadata, Exception>() {
+				@Override
+				public TopicMetadata doWithRetry(RetryContext context) throws Exception {
+					TopicMetadata topicMetadata = AdminUtils.fetchTopicMetadataFromZk(topicName, zkClient);
+					if (topicMetadata.errorCode() != ErrorMapping.NoError() || !topicName.equals(topicMetadata.topic())) {
+						// downcast to Exception because that's what the error throws
+						throw (Exception) ErrorMapping.exceptionFor(topicMetadata.errorCode());
+					}
+					List<PartitionMetadata> partitionMetadatas = new kafka.javaapi.TopicMetadata(topicMetadata).partitionsMetadata();
+					if (partitionMetadatas.size() != numPartitions) {
+						throw new IllegalStateException("The number of expected partitions was: " + numPartitions + ", but " +
+								partitionMetadatas.size() + " have been found instead");
+					}
+					for (PartitionMetadata partitionMetadata : partitionMetadatas) {
+						if (partitionMetadata.errorCode() != ErrorMapping.NoError()) {
+							throw (Exception) ErrorMapping.exceptionFor(partitionMetadata.errorCode());
+						}
+					}
+					return topicMetadata;
+				}
+			});
+			return topicMetadata;
+		}
+		catch (Exception e) {
+			String message = String.format("Cannot retrieve metadata for topic '%s'", topicName);
+			log.error(message, e);
+			throw new RuntimeException(message,e);
```

Comment:
Well, it may be redundant, but I'd rather have this logged here, in case the exception is swallowed. This is a serious error condition.